clients/fides-js/src/components/Overlay.tsx

```diff
@@ -108,12 +111,15 @@ const Overlay: FunctionComponent<Props> = ({
   });
 
   const handleOpenModal = useCallback(() => {
-    if (instance) {
+    if (options.fidesEmbed) {
+      setBannerIsOpen(false);
+      onOpen();
```

Comment:
Now that I think about it, this might fire twice 🤔

Reply:
yeah looks like currently we're firing this `onOpen` twice when fidesEmbed is enabled and we click into the modal. One fires here and one fires in a `useEffect` below- https://github.com/ethyca/fides/pull/4782/files#diff-12e4a1baa5e44f967e542dbb092fa1a6212d8ad7eec90b4b495c32dd707027edR132

We can probably remove the one here, since we need the `onOpen` to be handled in the case where we skip the banner, not only when we explicitly open the modal from the banner.

Reply:
Makes sense 👍 Yeah.. The state in this component is getting pretty complex (approaching truth table territory), but that's a project for another time.

Reply:
agreed. There's a lot of refactoring we can do with fides.js logic overall